### ISSUE TITLE
Allow specifying custom metadata when uploading Google Cloud Storage files

### ISSUE THREAD

**Issue by jannisy** [apache/pekko-connectors#656]
The [`resumableUpload`](url) in the Google Cloud Storage connector allows specifying metadata of type  Map[String, String] which are added to the uploaded file.
As per the [GCS documentation](https://cloud.google.com/storage/docs/json_api/v1/objects/insert), metadata properties like `contentType` or `cacheControl` are predefined by GCS. 

However, there is also _custom_ metadata, which is arbitrary user-provided key-value pairs. They are sent in the HttpRequest in the same way, but the value is not a string but a json object:
```javascript
{
    // metadata
    "contentType": "...",
    // custom metadata
    "metadata": {
        "customKey": "customData"
    }
}
```

However this cannot be modelled currently because metadata is Map[String, String] - the value needs to be specified as Map[String, String] instead of String to handle this.

(According to the documentatoin, GCS should also allow the notation `metadata.customKey = customData` but this did not work for me when I tried)

**Comment by jannisy** [comment]
Yes, makes sense. I will check again if I can find out what's the issue.

**Comment by mdedetrich** [comment]
This should apparently be working, there is a test for it at https://github.com/apache/pekko-connectors/blob/9f2ac5a9b54a655dd3350ae9830bc85c95c91d16/google-cloud-storage/src/test/scala/org/apache/pekko/stream/connectors/googlecloud/storage/impl/GCStorageStreamIntegrationSpec.scala#L267-L286

**Comment by mdedetrich** [comment]
So I actually ran the test suite against a proper GCS account and I can confirm that the current tests actually don't work, they seem to have either been improperly implemented or more likely GCS changed the behaviour around how metadata works.

```
[info] <-- [org.apache.pekko.stream.connectors.googlecloud.storage.impl.GoogleGCStorageStreamIntegrationSpec: GCStorageStream should provide a sink to stream data to gcs] End of log messages of test that [Failed(org.scalatest.exceptions.TestFailedException: None was not equal to Some(Map("meta-key-1" -> "value-1")))]
[info] - should provide a sink to stream data to gcs *** FAILED ***
[info]   None was not equal to Some(Map("meta-key-1" -> "value-1")) (GCStorageStreamIntegrationSpec.scala:285)
```

I will look into this
